Problem running test.py in raspberry and other linux

Hi everybody,
when i run this command in raspberry in python practice directory, i get the below error:
 sudo python -m unittest test.py

Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/usr/lib/python2.7/unittest/__main__.py", line 12, in <module>
    main(module=None)
  File "/usr/lib/python2.7/unittest/main.py", line 94, in __init__
    self.parseArgs(argv)
  File "/usr/lib/python2.7/unittest/main.py", line 149, in parseArgs
    self.createTests()
  File "/usr/lib/python2.7/unittest/main.py", line 158, in createTests
    self.module)
  File "/usr/lib/python2.7/unittest/loader.py", line 130, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/lib/python2.7/unittest/loader.py", line 100, in loadTestsFromName
    parent, obj = obj, getattr(obj, part)
AttributeError: 'module' object has no attribute 'py'



**If i run the test with python3 i get below error:**

sudo python3 -m unittest test.py 
FFFFFFFFFF
======================================================================
FAIL: testCase1 (test.test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/Desktop/rpi-course/python practice/day0/test.py", line 21, in testCase1
    self.assertEqual( result[0], result[1])
AssertionError: 'Traceback (most recent call last):\n  Fi[176 chars]al\n' != "Hello, World.\n'2#97^\n"
+ Hello, World.
- Traceback (most recent call last):
-   File "solution.py", line 1, in <module>
-     print("Hello, World.\n" + input())
-   File "<string>", line 1
-     '2#97^
? ----
+ '2#97^
-          ^
- SyntaxError: EOL while scanning string literal


======================================================================
FAIL: testCase10 (test.test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/Desktop/rpi-course/python practice/day0/test.py", line 57, in testCase10
    self.assertEqual( result[0], result[1])
AssertionError: 'Traceback (most recent call last):\n  Fil[153 chars]ax\n' != 'Hello, World.\nJ5[*/]\n'
+ Hello, World.
- Traceback (most recent call last):
-   File "solution.py", line 1, in <module>
-     print("Hello, World.\n" + input())
-   File "<string>", line 1
-     J5[*/]
? ----
+ J5[*/]
-        ^
- SyntaxError: invalid syntax

.
.
.
.
.
----------------------------------------------------------------------
Ran 10 tests in 1.104s

FAILED (failures=10)

--- a/python practice/day0/test.py
+++ b/python practice/day0/test.py
@@ -9,7 +9,7 @@ def string_generator(size=6, chars=string.ascii_uppercase + string.digits + " " 
 		
 def TestBase():
     stdIn = string_generator()
-    p = Popen(['python', 'solution.py'], stdout=PIPE, stdin=PIPE, stderr=STDOUT)   
+    p = Popen(['python3', 'solution.py'], stdout=PIPE, stdin=PIPE, stderr=STDOUT)   
     stdout = p.communicate(input=bytes(stdIn, encoding='utf-8'))[0]
     expectedStdout = "Hello, World." + os.linesep + stdIn + os.linesep
     return (stdout.decode(), expectedStdout)

